src/coreclr/jit/layout.cpp

```diff
@@ -36,34 +36,59 @@ class ClassLayoutTable
     unsigned m_layoutCount;
     // The capacity of m_layoutLargeArray (when more than 3 layouts are stored).
     unsigned m_layoutLargeCapacity;
+    // We furthermore fast-path the 0-sized block layout which is used for
+    // block locals that may grow (e.g. the outgoing arg area in every non-x86
+    // compilation).
+    ClassLayout m_zeroSizedBlockLayout;
 
 public:
-    ClassLayoutTable() : m_layoutCount(0), m_layoutLargeCapacity(0)
+    ClassLayoutTable() : m_layoutCount(0), m_layoutLargeCapacity(0), m_zeroSizedBlockLayout(0)
     {
     }
 
-    // Get the layout number (FirstLayoutNum-based) of the specified layout.
+    // Get a number that uniquely identifies the specified layout.
     unsigned GetLayoutNum(ClassLayout* layout) const
     {
+        if (layout == &m_zeroSizedBlockLayout)
+        {
+            return UINT_MAX;
+        }
+
         return GetLayoutIndex(layout) + FirstLayoutNum;
```

Comment:
`GetLayoutNum` initially was intended to be "small" (s.t. you could use it in bitsets for example), so I suppose it would be more consistent with that intent to use a `FirstLayoutNum`-based sentinel instead of `UINT_MAX`.

That said, we never ended up taking advantage of this property, so if you don't think it's important to preserve, no need to change it.